src/Runners/TestSuiteRunner.ts

```diff
@@ -3,15 +3,26 @@ import TestSuite from '../Framework/TestSuite';
 import TestSuiteResults from '../Framework/TestSuiteResults';
 import TestInfo from '../Framework/TestInfo';
 import TestRunner from './TestRunner';
+import TestName from '../Framework/TestName';
 
 export default class TestSuiteRunner {
 
     constructor(private readonly runner: TestRunner, private readonly reporters: ReadonlyArray<ResultReporter>) { }
 
-    async runSuite(suite: TestSuite) {
+    async runSuite(suite: TestSuite, testName?: string) {
```

Comment:
Do you have a preference between the `selected`/`all` branching in `Runner` and `TestSuiteLoader` vs the optional parameter here? We should probably keep the same pattern for consistency.

Reply:
My decision between the two approaches was really dependent on just how much additional logic a given method would need to perform the filtering. It makes sense to unify this approach though. Please see d61c47f for proposed resolution.